### PR TITLE
User should be able to sign up

### PR DESCRIPTION
**Pull request by ishimwebernard**
#### What does this PR do?
[what does this PR do?]'
>This PR allows users to sign up with email and password
> This PR allows the user to receive the verification email
>This PR allows a user to be verified if he/she only clicks the link
#### Description of Task to be completed
[the tasks that you completed on the feature]
>Setting up the model with email and password
>Using nodemailer to send email to the verification email
>Writting controllers and verifiers
image
image

### CODE REVIEW

**Comment by Edmondus** [comment]
@ishimwebernard can you update .env.example file?

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "start": "babel-node src/index.js",
+    "start": "babel-node src/index.js -- require @babel/plugin-proposal-class-properties",
```
Comment (mukunzidd) [inline package.json]: You don't need babel-node in production. The start script will be used to run the app in production.

File: .babelrc [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
     "presets": ["@babel/preset-env"],
-    "plugins": ["@babel/transform-runtime"]
+    "plugins": ["@babel/transform-runtime","@babel/plugin-proposal-class-properties"]
```
Comment (mukunzidd) [inline .babelrc:3]: @ishimwebernard for what purpose are you using "@babel/plugin-proposal-class-properties" module?

File: src/controllers/User.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+const SignUpValidator = require('../middlewares/validators/SignUpValidator');
+const user = require('../models/user');
+const jwt = require("jsonwebtoken");
+const nodemailer = require("nodemailer");
+const jwtDecode = require('jwt-decode');
+
+class User {
+    static signUp = async (request, response) => {
+        //Validation of data
+        // (async () => {
```
Comment (mukunzidd) [inline src/controllers/User.js]: Don't leave commented code in your PR. If you don't need this function then remove it.

File: src/routes/index.js [hunk 4]
```diff
@@ -1,15 +1,18 @@
 import { Router } from 'express';
 import swaggerUI from 'swagger-ui-express';
-import welcomeRoute from './welcome';
-import Badroute from './badroute';
+// import welcomeRoute from './welcome';
+// import Badroute from './badroute';
```
Comment (mukunzidd) [inline src/routes/index.js]: @ishimwebernard you shouldn't other people's features. This is a big no no... you should make your code increment work well with the already existing work.

File: src/routes/index.js [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
 
 router.use('/api-docs', swaggerUI.serve, swaggerUI.setup(swaggerDocs));
 
-router.use('/', welcomeRoute);
+router.use('/auth/', Auth);
 
-router.use('/', Badroute);
+// router.use('/', welcomeRoute);
+
+// router.use('/', Badroute);
```
Comment (mukunzidd) [inline src/routes/index.js]: @ishimwebernard  Same here. This is a big no no.

**Review by mukunzidd — CHANGES_REQUESTED** [review]
- Add documentation for this route
- Make sure the tests are passing. 
- Make sure that the user controller is tested well. Coverage should be at least 90%.
- Restore other people's work. The welcome route you commented out. 
- Address comments made on different files.

**Pull request merged**